implementations/rust/ockam/ockam_command/src/entry_point.rs

```diff
@@ -23,37 +24,46 @@ pub fn run() -> miette::Result<()> {
 
     let _ = miette::set_hook(Box::new(|_e| Box::new(ErrorReportHandler::new())));
 
-    match OckamCommand::try_parse_from(input.clone()) {
-        Err(help) => {
-            // the -h or --help flag must not be interpreted as an error
-            if !has_help_flag(&input) {
-                let command = input
-                    .iter()
-                    .take_while(|a| !a.starts_with('-'))
-                    .collect::<Vec<_>>()
-                    .iter()
-                    .map(|s| s.to_string())
-                    .collect::<Vec<String>>()
-                    .join(" ");
+    if has_version_flag(&input) {
+        let version_msg = Version::long();
+        let version_msg_vec = version_msg.split('\n').collect::<Vec<_>>();
+        eprintln!("{}", fmt_log!("ockam {}", version_msg_vec[0]));
+        for line_num in 1..version_msg_vec.len() {
+            eprintln!("{}", fmt_log!("{}", version_msg_vec[line_num]));
```

Comment:
Why do we need to handle the version flags ourselves (outside of clap)?

If have to do it, I'd avoid the if/else to improve the readability:

```rust
if has_version_flag(&input) {
    print_version_and_exit();
}

match OckamCommand::try_parse_from ...
```

Reply:
I still don't get it. Are we replacing [clap's version](https://github.com/build-trust/ockam/blob/aad7f0001a9c3fc7c0ef9cfd3d2ca77c7a028f3e/implementations/rust/ockam/ockam_command/src/command.rs#L38-L39) handling? Couldn't we use those attributes to define the function we want to call to display the version instead of manually handling the version arguments?

Reply:
if we use clap, we can't inject spaces before "ockam". we only control the string that comes after it. 

so in the display output, if we wanted, then we can use clap:
"ockam     version: XYZ"

however, since we want the spaces before the command name `ockam`, we have to override clap's behavior to get:
"    ockam version: XYZ"

Reply:
ooh! I get it now, thanks for showing an example, I didn't see that we wanted to add the margin before the "ockam" string. All clear now!

Reply:
@adrianbenavides Awesome! It is a tricky one that made no sense (to me either) w/out using that example :)